admin/test/features/generate_pupil_pins.feature

```diff
@@ -2,10 +2,34 @@
 Feature: Generate Pupil PINs
 
 
-  Background:
+  Scenario: Generate Pupil Landing page displays heading and info section
     Given I am logged in
-    And I am on the generate pupil pins page
-
-  Scenario: Add Multiple Pupil Landing page displays heading and sub heading section
+    When I navigate to generate pupil pins page
     Then I should see a heading for the Generate Pupil Pins
     And I can see the info message for generating the pupil pin
+
+  Scenario Outline: Generate Pupil List Page displays lists of Pupils
+    Given I have signed in with <teacher>
+    And I navigate to generate pupil pins page
+    When I click Generate PINs button
+    Then I should see a list of pupils sorted by surname on Generate Pins List Page
+
+  Examples:
+  | teacher  |
+  | teacher1 |
+  | teacher2 |
+  | teacher3 |
+  | teacher4 |
+
+  Scenario: Generate Pupil List Page do not display pupil with active pin
```

Comment:
@sinhaamit1980  does this page actually exist? "Generate Pupil List Page" ??

Reply:
Done changed it to Generate Pins Pupil List page